event-management-system/src/main/java/com/target/eventmanagementsystem/controller/EventController.java

```
@@ -18,8 +20,33 @@ public class EventController {
     private EventService eventService;
 
     @GetMapping("/getAllEvents")
-    public List<Events> list(){
-        return eventService.listall();
+    public ResponseEntity<ApiResponse<Object>> list(){
+
+        try
+        {
+            Object responseData = eventService.listall();
+
+            // Create a success response
+            ApiResponse<Object> response = new ApiResponse<>();
+            response.setStatusCode(HttpStatus.OK.value());
+            response.setStatusMessage("Success");
+            response.setData(responseData);
+
+            return ResponseEntity.ok(response);
+        }catch (Exception e) {
+            // Create an error response
+            ApiError error = new ApiError();
```

I think it is not a good idea to handle the exceptions this way. 
It is better to have a custom exception class like below and use Spring ControllerAdvice to handle that. 

```
NewExceptionClass extends RuntimeException {
   HttpStatusCode statusCode;
   String errorMessage;
   //Add more fields if required
}
```

And you can throw this exception class when ever you catch an exception that hinders the flow.
Like if you are querying DB and caught an exception. You can catch that exception and convert it into this new exception.

And in the controllerAdvice you will catch this exception and return a responseEntity object like below

 ```
handleNewException(NewExceptionClass ex) {
     //Create APIResponse and APIError class using the ex.getErrorMessage.
     return new ResponseEntity<>(ex.getStatusCode, ApiResponse)
}
```